packages/fuels/src/cli/commands/dev/index.ts

```diff
@@ -18,7 +18,24 @@ export const closeAllFileHandlers = (handlers: FSWatcher[]) => {
 
 export const buildAndDeploy = async (config: FuelsConfig) => {
   await build(config);
-  return deploy(config);
+  config.onSuccess?.(
+    {
+      type: Commands.build,
+      data: null,
+    },
+    config
+  );
```

Comment:
Probably a good idea to move this inside the `build` and `deploy` methods.

Also, we should consider removing the `onSucces` in favor of `onBuild`, `onDeploy`, etc.

Reply:
What if we removed the `onSucces` in favor of `onBuild`, `onDeploy`, etc.?

Reply:
@arboleya that would cover the `build`, `deploy` and `dev` commands. Do we have more commands that use `onSuccess` to your knowledge?

Reply:
Only if we want to support the [other commands](https://github.com/FuelLabs/fuels-ts/tree/master/packages/fuels/src/cli/commands):
 - `init` / onInit?
 - `node` / onNode?

There could also be an `onDev` event that fires up every time a hot reload occurs, for example. So it would look like:

 1. `onBuild`
 1. `onDeploy`
 1. `onDev`

Reply:
I would be in favour of supporting all of them.

Reply:
And getting rid of `onSuccess` if we add all of these (?) @arboleya

Reply:
Sounds like a plan. 👌

Reply:
Wonderful. 🙌

Reply:
Yeah, I prefer the second option as well.